### PR TITLE
:sparkles: Add compilation time test

### PR DESCRIPTION
**Pull request by gkumbhat**

### CODE REVIEW

File: aiu_fms_testing_utils/utils/__init__.py [hunk 1]
```diff
@@ -67,6 +67,19 @@ def warmup_model(
     dprint(f"PT compile complete, took {pt_compile_model_time:.3f}s")
 
 
+def get_env_to_int_list(env_var_name, default):
```
Comment (JRosenkranz) [inline aiu_fms_testing_utils/utils/__init__.py:70]: can you add type hints to this as well as explanation for each parameter

Comment (JRosenkranz) [inline aiu_fms_testing_utils/utils/__init__.py:70]: I could see this function serving much more utility than just ints. I think we may want to include a third optional parameter which is a function which takes the value and returns the value the user wants. (by default it would just be a list of strings). For instance, we may want to return a list of floats or custom objects.

File: tests/testing/test_compilation.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    warmup_model,
+)
+from aiu_fms_testing_utils.utils.aiu_setup import dprint
+
+GRANITE_3p3_8B_INSTRUCT = "ibm-granite/granite-3.3-8b-instruct"
+SHARE_GPT_DATASET_PATH = os.environ.get(
+    "SHARE_GPT_DATASET_PATH", os.path.expanduser("~/share_gpt.json")
+)
+
+ATTN_NAME = "spyre_paged_attn"
```
Comment (JRosenkranz) [inline tests/testing/test_compilation.py:30]: we should make this configurable

File: tests/testing/test_compilation.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+from aiu_fms_testing_utils.utils import (
+    ids_for_prompt,
+    get_env_to_int_list,
+    sample_sharegpt_requests,
+    warmup_model,
+)
+from aiu_fms_testing_utils.utils.aiu_setup import dprint
+
+GRANITE_3p3_8B_INSTRUCT = "ibm-granite/granite-3.3-8b-instruct"
+SHARE_GPT_DATASET_PATH = os.environ.get(
```
Comment (JRosenkranz) [inline tests/testing/test_compilation.py:26]: given this is just testing compilation time, I don't think we need real data

File: tests/testing/test_compilation.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    "SHARE_GPT_DATASET_PATH", os.path.expanduser("~/share_gpt.json")
+)
+
+ATTN_NAME = "spyre_paged_attn"
+
+COMPILE_DYNAMIC_SHAPE = True
```
Comment (JRosenkranz) [inline tests/testing/test_compilation.py:32]: we should make this configurable

File: tests/testing/test_compilation.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+common_batch_sizes = get_env_to_int_list("FMS_TEST_SHAPES_COMMON_BATCH_SIZES", [1])
+common_seq_lengths = get_env_to_int_list("FMS_TEST_SHAPES_COMMON_SEQ_LENGTHS", [64])
+common_max_new_tokens = get_env_to_int_list(
+    "FMS_TEST_SHAPES_COMMON_MAX_NEW_TOKENS", [64]
+)
+common_expected_comp_time = get_env_to_int_list(
+    "COMMON_COMPILATION_EXPECTED_TIME", [10]
+)  # In minutes
+
+COMMON_SHAPE_TYPE = "dynamic"
```
Comment (JRosenkranz) [inline tests/testing/test_compilation.py:45]: we should make this configurable

File: tests/testing/test_compilation.py [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+        common_batch_sizes,
+        common_seq_lengths,
+        common_max_new_tokens,
+        common_expected_comp_time,
+    )
+)
+
+
+# TODO: This is copied from test_decoders.py would be good to consolidate
+def __prepare_inputs(batch_size, seq_length, tokenizer, seed=0):
```
Comment (JRosenkranz) [inline tests/testing/test_compilation.py:74]: we can make this simpler and just use a simple `torch.arange` with the sizes required

File: tests/testing/test_compilation.py [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+)
+def test_compilation_time(
+    model_path, batch_size, seq_length, max_new_tokens, expected_comp_time
+):
+    """Test to validate time taken for model compilation."""
+    torch.manual_seed(42)
+    torch.set_default_dtype(torch.float16)
+    os.environ["COMPILATION_MODE"] = "offline_decoder"
+
+    dprint(
```
Comment (JRosenkranz) [inline tests/testing/test_compilation.py:111]: include attention type to this print

File: tests/testing/test_compilation.py [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+    extra_kwargs["attn_name"] = ATTN_NAME
+
+    start_time = time.perf_counter()
+    if COMMON_SHAPE_TYPE == "dynamic":
+        COMPILE_DYNAMIC_SHAPE = True
+    else:
+        COMPILE_DYNAMIC_SHAPE = False
+
+    model.compile(backend="sendnn", options={"sendnn.dynamic": COMPILE_DYNAMIC_SHAPE})
+    warmup_model(
```
Comment (JRosenkranz) [inline tests/testing/test_compilation.py:145]: This only includes the initial compile (not the device warmup). Do we want to include that as well (inference.py has an example https://github.com/foundation-model-stack/aiu-fms-testing-utils/blob/bd1090ef7aa1323b28587533d29935827b7a6755/scripts/inference.py#L824)
